### ISSUE TITLE
When using jsx: react-jsx in tsconfig.json, react key props error occurs

### ISSUE THREAD

**Issue by chebread** [oven-sh/bun#2049]
### What version of Bun is running?

0.5.6

### What platform is your computer?

Darwin 22.3.0 x86_64 i386

### Bugs

I am using typescript based react with bun create react.
I have configured jsx: react-jsx in tsconfig.json.

However, a `Warning: Each child in a list should have a unique "key" prop.` error occurred in the console.
This error occurs when there are two or more tags (eg div tags).

However, this error is solved by using jsx: react-jsxdev.

Can anyone help me resolve this error?

**Comment by akomm** [comment]
This input:
```tsx
const App: FC = () => (
  <div>
    <div>foo</div>
    <pre>bar</pre>
  </div>
)
```
Results in this output:
```ts
var App = () => jsx_runtime.jsx("div", {
  children: [
    jsx_runtime.jsx("div", {
      children: "foo"
    }),
    jsx_runtime.jsx("pre", {
      children: "bar"
    })
  ]
});
```

The problem is the children array at the root `div`. At least if its passed like that further to react jsx.. 
Typically this output (notice the absence of array) would be generated in react:

```tsx
React.createElement("div", null,
  React.createElement("div", null, "foo"),
  React.createElement("pre", null, "bar")
);
```

We only get array when transpiling this:

```tsx
var App: FC = () => <div>
  {[
    <div>foo</div>,
    <pre>bar</pre>
  ]}
</div>
```

You would get children as array:

```tsx
React.createElement("div", null,
  [
    React.createElement("div", null, "foo"),
    React.createElement("pre", null, "bar")
  ]
);
```

**Comment by vjpr** [comment]
I was getting red squiggles in IntelliJ/WebStorm. Needed to set `"jsx": "react"`.

**Comment by akomm** [comment]
Just confirming that it appears not correct and where the problem is. Yes if you switch back to the "older" compilation mode its output correct and you get no warning. But using the old isn't "desired".

**Comment by vjpr** [comment]
I had forgot to install `typescript` in my monorepo root.

**Comment by dylan-conway** [comment]
#2968 might fix this issue

**Comment by Electroid** [comment]
Closed by #2968

**Comment by bryanjtc** [comment]
I still have the problem. Using bun 1.0.15
